### PR TITLE
Show task history in the admin section

### PR DESCRIPTION
**Pull request by kdoh**
Frontend display the task history that began in #8397 to show tasks MB is running to help users debug performance and other issues.

TODOs
- [x] API endpoint to return task list
- [x] API endpoint to return individual task
- [x] Tests for `api/task` endpoints (probably @senior)
- [x] Replace stub with actual api list from endpoint
- [x] Modal route to show details of a particular task
- [x] Come up with a better name than `Task` ?
- [x] Handle paginated task history endpoint
  - [x] pagination ui - (@kdoh)
  - [x] pagination logic  - (@tlrobinson)

### CODE REVIEW

File: src/metabase/api/task.clj [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    [metabase.models
+     [task-history :as tasks :refer [TaskHistory]]]
+    [toucan
+     [db :as db]]))
+
+
+(api/defendpoint GET "/"
+  "Fetch a list of recent tasks stored as Task History"
+  []
+  (as-> (db/select TaskHistory) tasks))
```
Comment (senior) [inline src/metabase/api/task.clj]:
```suggestion
  (api/check-superuser)
  (db/select TaskHistory)
```

File: src/metabase/api/task.clj [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+(api/defendpoint GET "/:id"
+  "Get `TaskHistory` entry with ID."
+  [id]
+  (u/prog1 (-> (TaskHistory id)
+               api/read-check)))
```
Comment (senior) [inline src/metabase/api/task.clj]: Looks like I can't "suggest" with two lines. What you have will work, but this is an easier way to say it:
```
(api/read-check TaskHistory id)
```
That will check for existence, permissions and query the DB.

File: src/metabase/api/task.clj [hunk 3]
```diff
@@ -0,0 +1,26 @@
+(ns metabase.api.task
+  "/api/task endpoints"
+  (:require
+    [compojure.core :refer [GET]]
+    [metabase
```
Comment (senior) [inline src/metabase/api/task.clj]: When there's just one namespace, you are importing, you can use:
```
[metabase.util :as u]
```
Separating it like you're doing here used more when you're importing more than one, i.e.
```
[metabase
  [schema :as s]
  [util :as u]]
```

File: src/metabase/api/task.clj (hunk 1 above)
Comment (senior) [reply src/metabase/api/task.clj]: This will also fix the linter error in this code.

**Review by senior — CHANGES_REQUESTED** [review]
Seems pretty close, happy to jump in and help if you need it @kdoh .

**Comment by mazameli** [comment]
Re: what to call this tab/page, maybe `Troubleshooting`? I.e., in the future we can envision putting additional logs here, such as ones that detail information about timezone-related things to help troubleshoot TZ issues.

If we go with that label, I think the page will need some context-setting text at the top, like "Trying to get to the bottom of something? This section shows logs of Metabase's background tasks, which can help shed light on what's going on."

(This could then also be a place where we could consider including a link to our paid support info page.)

**Comment by kdoh** [comment]
Yeah I think in terms of the long term vision for what this section of the admin section will do I dig `Troubleshooting` as the overall section heading. I went ahead and updated the nav item and urls to start. I'll fix up the header and add some context here in a little.

**Comment by mazameli** [comment]
The details modal content can sometimes overflow. Seems like we'd want it to wrap.

overflow

**Comment by mazameli** [comment]
It'd be great if these timestamps were formatted in a friendlier way.

screen shot 2018-10-17 at 3 48 05 pm

**Comment by mazameli** [comment]
Minor annoyance: if I open up the details modal, it'll scroll me to the top of the page if I click outside the modal to close it. If I click the X to close it, I stay at my scroll position as expected:

scrolls-to-top

**Comment by mazameli** [comment]
If I'm being super duper nit-picky, I'd expect the colors here to match or for the info icon to be lighter than the heading I think, and the icon to be scooted down a little to align it with the heading.

screen shot 2018-10-17 at 3 55 55 pm

**Comment by mazameli** [comment]
Could we relatively easily do a remapping so that the `DB ID` column displays the name of the database instead?

screen shot 2018-10-17 at 4 00 33 pm

**Comment by mazameli** [comment]
Cool, as discussed, I'm a 👍on merging this, with the timestamp and DB ID formatting changes coming later as time allows as follow-ons. I'll create a task history polish issue.

**Pull request merged**